app/models/service/aggregation.rb

```diff
@@ -42,26 +273,51 @@ def aggregate_direct_vm_memory_on_disk
   end
 
   def aggregate_all_vm_cpus
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.cpu_total_cores.to_i }
+    if has_attribute?("aggregate_all_vm_cpus")
+      self["aggregate_all_vm_cpus"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.cpu_total_cores.to_i }
+    end
   end
 
   def aggregate_all_vm_memory
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.ram_size.to_i }
+    if has_attribute?("aggregate_all_vm_memory")
+      self["aggregate_all_vm_memory"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.ram_size.to_i }
+    end
   end
 
   def aggregate_all_vm_disk_count
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.num_disks.to_i }
+    if has_attribute?("aggregate_all_vm_disk_count")
+      self["aggregate_all_vm_disk_count"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.num_disks.to_i }
+    end
   end
 
   def aggregate_all_vm_disk_space_allocated
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.allocated_disk_storage.to_i }
+    if has_attribute?("aggregate_all_vm_disk_space_allocated")
+      self["aggregate_all_vm_disk_space_allocated"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.allocated_disk_storage.to_i }
+    end
   end
 
   def aggregate_all_vm_disk_space_used
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.used_disk_storage.to_i }
+    if has_attribute?("aggregate_all_vm_disk_space_used")
+      self["aggregate_all_vm_disk_space_used"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.used_disk_storage.to_i }
+    end
   end
 
   def aggregate_all_vm_memory_on_disk
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.ram_size_in_bytes_by_state.to_i }
+    if has_attribute?("aggregate_all_vm_memory_on_disk")
+      # Avoids (poorly) "ERROR:  integer out of range" in postgres
+      self["aggregate_all_vm_memory_on_disk"] * 1.megabyte
```

Comment:
So in re-review, this comment concerns me.  This implies that the math is being done on ints instead of bigint.  Can we just cast to bigint?